src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -272,22 +276,49 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (contain_mutable_functions((Node *) targetEntry->expr))
+			if (commandType == CMD_UPDATE &&
+				contain_volatile_functions((Node *) targetEntry->expr))
```

Comment:
Ping? Shouldn't we run similar checks for DELETE (albeit not on the targetlist)?